inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java

```diff
@@ -0,0 +1,222 @@
+    public static final String JOB_MQTT_QOS = "job.mqttJob.qos";
+    public static final String JOB_MQTT_CLEAN_SESSION = "job.mqttJob.cleanSession";
+    public static final String JOB_MQTT_CLIENT_ID_PREFIX = "job.mqttJob.clientIdPrefix";
+    public static final String JOB_MQTT_QUEUE_SIZE = "job.mqttJob.queueSize";
+    public static final String JOB_MQTT_AUTOMATIC_RECONNECT = "job.mqttJob.automaticReconnect";
+    public static final String JOB_MQTT_VERSION = "job.mqttJob.mqttVersion";
+
+    private boolean finished = false;
+
+    private boolean destroyed = false;
+
+    private MqttClient client;
+
+    private MqttConnectOptions options;
+
+    private String serverURI;
+    private String userName;
+    private String password;
+    private String topic;
+    private int qos;
+    private boolean cleanSession = false;
+    private boolean automaticReconnect = true;
+    private JobProfile jobProfile;
+    private String instanceId;
+    private String clientId;
+    private int mqttVersion = MqttConnectOptions.MQTT_VERSION_DEFAULT;
+
+    private LinkedBlockingQueue<DefaultMessage> mqttMessagesQueue;
+
+    public MqttReader(String topic) {
+        this.topic = topic;
+    }
+
+    @Override
+    public void init(JobProfile jobConf) {
+        super.init(jobConf);
+        jobProfile = jobConf;
+        LOGGER.info("init mqtt reader with jobConf {}", jobConf.toJsonStr());
+
+        mqttMessagesQueue = new LinkedBlockingQueue<>(jobConf.getInt(JOB_MQTT_QUEUE_SIZE, 1000));
+        instanceId = jobConf.getInstanceId();
+        userName = jobConf.get(JOB_MQTT_USERNAME);
+        password = jobConf.get(JOB_MQTT_PASSWORD);
+        serverURI = jobConf.get(JOB_MQTT_SERVER_URI);
+        clientId = jobConf.get(JOB_MQTT_CLIENT_ID_PREFIX, "mqtt_client") + "_" + UUID.randomUUID();
+        cleanSession = jobConf.getBoolean(JOB_MQTT_CLEAN_SESSION, false);
+        automaticReconnect = jobConf.getBoolean(JOB_MQTT_AUTOMATIC_RECONNECT, true);
+        qos = jobConf.getInt(JOB_MQTT_QOS, 1);
+        mqttVersion = jobConf.getInt(JOB_MQTT_VERSION, MqttConnectOptions.MQTT_VERSION_DEFAULT);
+
+        options = new MqttConnectOptions();
+        options.setCleanSession(cleanSession);
+        options.setConnectionTimeout(jobConf.getInt(JOB_MQTT_CONNECTION_TIMEOUT, 10));
+        options.setKeepAliveInterval(jobConf.getInt(JOB_MQTT_KEEPALIVE_INTERVAL, 20));
+        options.setUserName(userName);
+        options.setPassword(password.toCharArray());
+        options.setAutomaticReconnect(automaticReconnect);
+        options.setMqttVersion(mqttVersion);
+
+        try {
+            synchronized (MqttReader.class) {
+                client = new MqttClient(serverURI, clientId, new MemoryPersistence());
+                client.setCallback(new MqttCallback() {
+                    @Override
+                    public void connectionLost(Throwable cause) {
+                        LOGGER.info("the mqtt connection is lost, try to reconnect");
```

Comment:
It's better to add  some necessary info such as jobId, connection etc in this log.